### DISCUSSION TITLE
Best practice to use Coolify for local development

### DISCUSSION THREAD

**Discussion by digitalit** [coollabsio/coolify#2128] [Q&A]
Hi guys,

I really like Coolify and i've starting to use it for production with Hetzner and i would like to use it for development on my Proxmox server.

I got it running like in Hetzner/production on a Proxmox LXC and attached separate Proxmox LXC project servers.

Now here is what i guess things differs from an ordinary set up. Sure i want to deploy/get my repo from github but only once. Nothing should redeploy on commits as this is the development code and it will get committed to main branch and then deployed on the Hetzner production Coolify server.

I also don't want to work in the web version of VSCode (code-server), i want to remote connect to the project/docker/code over SSH.

As i'm not that skilled with Docker my idea was to manually install like Node.js and what not on the LXC and just use it like i already do but use Coolify to create a Postgres project and connect to it directly. This way i will keep my VSCode remote dev experience but i can use Coolify's stuff for DB storage/backup.
 
This might be stupid and incorrect way of using Coolify for development but i'd like your input on this please.

Have a nice day and a big thank you to the creator of Coolify.

**Comment by fcpauldiaz** [comment]
I think your approach is reasonable, don't forget to add a firewall to your instance. If you want to use a full dev server, you can use nixpacks as deployment and disable automatic deploys from GitHub

**Comment by peaklabs-dev** [comment]
@digitalit Sounds good to me. I think this can be closed, if not please re-open it.
